src/main/java/hudson/plugins/git/UserRemoteConfig.java

```diff
@@ -127,7 +129,7 @@ public FormValidation doCheckCredentialsId(@AncestorInPath Item project,
             return FormValidation.ok();
         }
 
-        public FormValidation doCheckUrl(@AncestorInPath Item project,
+        public FormValidation doCheckUrl(@AncestorInPath Job project,
```

Comment:
Why not just do an instanceof check and fall back to System.getEnv where it's an Item but not a Job?